### PR TITLE
beta tests for kotlin

### PR DESCRIPTION
**Pull request by colleend**
Adding all the `semgrep-core` tests necessary for beta!

* all tests #justworked except for `metavar_equality_stmt`, which i filed this bug: https://github.com/returntocorp/semgrep/issues/4234 for.

### CODE REVIEW

**Review by brendongo — APPROVED** [review]

File: semgrep-core/tests/kotlin/metavar_equality_stmt.sgrep [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if ($E)
```
Comment (aryx) [inline semgrep-core/tests/kotlin/metavar_equality_stmt.sgrep:1]: so it works then?

Comment (colleend) [reply semgrep-core/tests/kotlin/metavar_equality_stmt.sgrep:1]: yep!

**Comment by aryx** [comment]
Oops, those tests were actually never executed. We forgot code in  tests/Test.ml to run those tests.

**Comment by aryx** [comment]
I actually got some regressions today spotted by semgrep-rules, but not by semgrep-core tests because they were never executed.

**Comment by colleend** [comment]
🤦 totally forgot about that, thanks for adding it in!

**Pull request merged**